rs/embedders/src/wasm_executor.rs

```diff
@@ -705,7 +705,10 @@ pub fn process(
         if execution_parameters.instruction_limits.slicing_enabled()
             && dirty_pages.get() > embedder.config().max_dirty_pages_without_optimization as u64
         {
-            if let Err(err) = system_api.yield_for_dirty_memory_copy(instruction_counter) {
+            // Passing zero instructions here means that heavy memory execution consumed
+            // all the round time, and the round should be finished now. It does not affect
+            // the actual canister charges.
+            if let Err(err) = system_api.yield_for_dirty_memory_copy(0) {
```

Comment:
While this fixes the issue, I wonder if we should instead change the API to not accept `instruction_counter` at all. The only use case is here and the correct thing to do is to not allow any more instructions to be executed. So, I'd remove the parameter and set to 0 in the implementation in the system API. This seems to capture better the intent of what this `yield_for_dirty_memory_copy` should do imo.